vertx-sql-client-template/src/main/java/io/vertx/sqlclient/template/impl/SqlTemplate.java

```diff
@@ -0,0 +1,50 @@
+package io.vertx.sqlclient.template.impl;
+
+import io.vertx.sqlclient.Tuple;
+import io.vertx.sqlclient.impl.SqlClientInternal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SqlTemplate {
+
+  private static Pattern PARAM_PATTERN = Pattern.compile(":(\\p{javaUnicodeIdentifierStart}\\p{javaUnicodeIdentifierPart}*)");
```

Comment:
This will not work for postgresql :: cast

This is how spring does it
https://github.com/spring-projects/spring-framework/blob/4ca64684f6d5b7414ca3628a6b372a7c586fd2a6/spring-jdbc/src/main/java/org/springframework/jdbc/core/namedparam/NamedParameterUtils.java#L82

Another solution is to use ${name} via this pattern

```
Pattern.compile("\\$\\{\\s*(.*?)\\s*}");
```

We could also cache the result

Reply:
I like this idea